File: pinyougou/pinyougou-seckill/pinyougou-seckill-interface/src/main/java/com/pinyougou/seckill/service/SeckillGoodsService.java
package com.pinyougou.seckill.service;

import com.pinyougou.pojo.TbSeckillGoods;
import com.pinyougou.service.BaseService;
import com.pinyougou.vo.PageResult;

import java.util.List;

public interface SeckillGoodsService extends BaseService<TbSeckillGoods> {

    PageResult search(Integer page, Integer rows, TbSeckillGoods seckillGoods);

    /**
     * 查询全部秒杀商品,首页展示
     * @return
     */
    List<TbSeckillGoods> findList();

    /**
     * 根据id从redis中查询秒杀商品
     * @param id
     * @return
     */
    TbSeckillGoods findOneFromRedis(Long id);
}

File: pinyougou/pinyougou-seckill-web/src/main/webapp/js/service/seckillGoodsService.js
//定义业务服务
app.service("seckillGoodsService",function ($http) {

    this.getUsername=function () {
        return $http.get("seckillOrder/getUsername.do");
    };

    this.findList = function () {
        return $http.get("../seckillGoods/findList.do");
    };

    this.findOne = function (id) {
        return $http.get("../seckillGoods/findOne.do?id=" + id);
    };

    this.submitOrder = function (id) {
        return $http.get("../seckillOrder/submitOrder.do?seckillId=" + id);
    };
});